src/Entity/Site.php

```diff
@@ -95,11 +103,52 @@ public function removeDomain(Domain $domain): self
             }
         }
 
+        $this->setPrimaryDomain();
+
         return $this;
     }
 
     public function __toString(): string
     {
         return $this->configFilePath;
     }
+
+    public function getInstallation(): ?Installation
+    {
+        return $this->installation;
+    }
+
+    public function setInstallation(?Installation $installation): self
+    {
+        $this->installation = $installation;
+
+        return $this;
+    }
+
+    public function getPrimaryDomain(): ?string
+    {
+        return $this->primaryDomain;
+    }
+
+    private function setPrimaryDomain(): self
```

Comment:
Set method that do not take parameters? That requires some documentation?

Reply:
Renamed